### ISSUE TITLE
Partial Copy Sandbox: When parent object isn't imported

### ISSUE THREAD

**Issue by camisotro** [afawcett/declarative-lookup-rollup-summaries#437]
I have a situation where I created a Partial Copy Sandbox of a client instance and Salesforce for some reason sampled some child objects without their parent.

It brought in 10,000 Work Orders and 10,000 Services (this is a Field Service Lightning Managed Package object). About half the Services did not have their parent WO in place but the ID remains in the ID field. As a result DLRS threw a fit when I tried to update or delete any of those Services. For that matter it really botched the Work Order sampling... only 45 of 10,000 had their parent Case imported. This is really strange because my last Partial Copy, before enabling a DLRS, did not have anything close to this many anomalies. I wonder if something in DLRS is causing the sampling process to have issues.

Is it possible to change behaviour or provide an option for it, so that if the ID in a lookup field appears to not actually exist, DLRS gracefully ignores the rollup?

**Comment by afawcett** [comment]
Yeah for sure, i'll mark this as an enhancement to think about, sounds like a good valid use case. In the meantime, you can use the Active checkbox on the rollups to deactivate. Then use Calculate button to catchup, then re-Activate. Would that work?

**Comment by camisotro** [comment]
The problem is that Salesforce is allowing invalid IDs to live in
relationship fields, and DLRS always trusts that the IDs are valid. As far
as I can tell, SF does allow resaving the record without changing the
invalid ID. But any attempt to update a child that points to an invalid
parent will throw an error if DLRS tries to recalculate on a nonexistent
parent. So I don't think your suggestion solves it. After reactivating the
rule, every Service with an invalid parent would still throw an error upon
edit.

**Comment by afawcett** [comment]
Ok so... this https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/259 might be what we are talking about then in that case. Have a read let me know.

**Comment by camisotro** [comment]
Not totally clear whether that'd do the trick because I haven't inspected
your code. The scenario is, the child trigger runs and the parent ID is
invalid. Presumably your code has to run an aggregate query of all children
sharing the same parent ID, retrieves the aggregate value, then constructs
a SObject for the parent and does a DML update of the rollup field. If the
parent ID is invalid, this is where it would throw. Does that sound about
right?

**Comment by afawcett** [comment]
Yeah that does sound about it. I think this will be covered by https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/259 tbh. I'll leave this issue open though, once 259 is fixed, we can retest.

**Comment by camisotro** [comment]
Just wanted to add that I'm running into this again in another partial copy sandbox. Same type of scenario except different objects. Child object's lookup to the parent is an invalid ID. Can't modify the child at all without throwing a DLRS error.

**Comment by camisotro** [comment]
Yeah, partial sandbox anomalies cause all kinds of weirdness. If I develop code in a partial sandbox or plan on deploying to one, I add normally unnecessary workarounds like querying on myLookup__r.id != null instead of just myLookup__c != null.
